node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala

```diff
@@ -45,6 +57,98 @@ case class NeutrinoNode(
     this
   }
 
+  override def getPeersFromConf: Vector[Peer] = {
+    if (confPeersOverride.isEmpty) super.getPeersFromConf
+    else confPeersOverride
+  }
+
+  override def handlePeerGossipMessage(message: GossipAddrMessage): Unit = {
+    message match {
+      case addr: AddrMessage =>
+        addr.addresses.foreach(addToPeerQueue)
+      case addr: AddrV2Message =>
+        val bytes = addr.bytes
+        val port = addr.port.toInt
+        val services = ServiceIdentifier.fromBytes(addr.services.bytes)
+        val inetAddress =
+          NetworkUtil.parseInetSocketAddress(bytes, port)
+        val peer = Peer.fromSocket(socket = inetAddress,
+                                   socks5ProxyParams =
+                                     nodeAppConfig.socks5ProxyParams)
+        addr match {
+          case IPv4AddrV2Message(_, _, _, _) | IPv6AddrV2Message(_, _, _, _) =>
+            if (services.nodeCompactFilters)
+              peersToCheckStack.push(peer)
+          case TorV3AddrV2Message(_, _, _, _) =>
+            if (nodeConfig.torConf.enabled && services.nodeCompactFilters)
+              peersToCheckStack.push(peer)
+          case _ => logger.debug("Unsupported network. Skipping.")
+        }
+    }
+  }
+
+  override def onPeerInitialization(peer: Peer): Future[Unit] = {
```

Comment:
I think this should be moved into the `ControlMessageHandler` i talked about above. 

Also, more generally, there are some race conditions in this code. You need to be very careful about not dropping futures on the floor like you are doing with

```peerData(peer).peerMessageSender.sendGetAddrMessage()```

In your currently implementation, that `Future[Unit]` returned by this method is dropped on the floor, this means if an exception is caused by the method `sendGetAddrMessage()` , the caller of this method (`onPeerInitialization()`) will not have the exception propogated up the stack. It will silently fail which makes it _really_ hard to debug.

I think this could be re-written as
```scala
    if (
      peerData.contains(peer) && peerData(
        peer).serviceIdentifier.nodeCompactFilters
    ) {
  
      val createdInDbF = createInDb(peer)
      val handlePeerF = if (
        peerData(
          peer).keepConnection || connectedPeersCount < maxConnectedPeers
      ) {
        val sendMsgF = peerData(peer).peerMessageSender.sendGetAddrMessage()
        //only the nodes that have keepConnection as true would be actually used by us
        peerData(peer).keepConnection = true
        logger.info(
          s"Connected to peer $peer with compact filters. Connected peer count $connectedPeersCount")
        sendMsgF
      } else {
        removePeer(peer)
        Future.unit
      }
  for {
        _ <- createdInDbF
       _ <- handlePeerF
      } yield {
    ()
      }
    } else {
      removePeer(peer)
      Future.unit
    }

```

Sorry for the poor formatting, hopefully the idea gets across.